### ISSUE TITLE
readthedocs sphinx autodoc finufft module import fails

### ISSUE THREAD

**Issue by ahbarnett** [flatironinstitute/finufft#143]
We currently, as of v2.0.0, have broken autodoc sphinx doc web page for python docstrings.
See bottom part of:
https://finufft.readthedocs.io/en/latest/python.html
The docstrings are absent!

I thought this was because py 2.7 was being used by readthedocs. However I fixed up a .readthedocs.yml to specify py 3.7. But it has not fixed it, although it changed the error.
But still autodoc cannot import the finufft module.
RTD logs show:
```
WARNING: autodoc: failed to import module 'finufft'; the following exception was raised:
No module named 'finufft._finufft'
```
Eg https://readthedocs.org/api/v2/build/12034459.txt

This seems to imply it is trying to `import finufft` while in the `finufft/python` dir, 
but is consistent with two types of problem: either
1) it never pip installed the module, or
2) I notice that `import finufft` fails even though the module exists from pip, when run from the finufft/python dir, because it gets confused about the dir (not module) named finufft?

Here is what is supposed to happen to build the py module, in order that Joakim's fancy docstrings actually get generated:
```
	(export FINUFFT_DIR=$(shell pwd); cd python; $(PYTHON) -m pip install .)
```
That's taken from the makefile. (Note this is new since v2.) The necessity of the FINUFFT_DIR env var is only to build the _finufft extension (linking to the C++ lib, etc). But surely that is not needed for the docs, and Dylan had already used autodoc_mock_imports to prevent errors in _finufft from stopping the import.

I have attempted in conf.py to set this env var. But I can't figure out if it's working.

Does RTD know to pip install the module (rather than just read it) ?  The RTD logfile is not very informative.

As a python non-user I simply don't know enough about pip or RTD to debug this,
so after a couple of stuck hours, I'm asking @dylex @janden @lu1and10 for help.

**Comment by janden** [comment]
> I notice that import finufft fails even though the module exists from pip, when run from the finufft/python dir, because it gets confused about the dir (not module) named finufft?

Yes I've noticed this as well. You have to step out of the directory for things to work. I'm not sure why that is.

> But surely that is not needed for the docs, and Dylan had already used autodoc_mock_imports to prevent errors in _finufft from stopping the import.

Ok that makes sense. I wonder why the import errors in that case.

> Does RTD know to pip install the module (rather than just read it) ?

This I'm fairly sure is necessary. I'm less familiar with RTD, but sphinx *must* import the module to read the docstrings, so presumably this must be the case for RTD as well, since it uses sphinx here.

I'll play around with this and see what we can do to get things working.

**Comment by ahbarnett** [comment]
fixed by #143 (I was so close...)
